### PR TITLE
WooCommerce: Add selectors for last wcApi error.

### PR DESCRIPTION
**Pull request by coderkevin**
This adds selectors for the error object, the code, and the message (if
available).

To Test:

`make test` and ensure tests run successfully.

### CODE REVIEW

File: client/extensions/woocommerce/state/site/status/wc-api/selectors.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+export function getLastApiError( rootState, siteId ) {
+	const woocommerce = rootState.extensions.woocommerce;
+	return get( woocommerce, 'site[' + siteId + '].status.wcApi.error.data' );
```
Comment (belcherj) [inline client/extensions/woocommerce/state/site/status/wc-api/selectors.js]: I don't love this, maybe string interpolation would be nicer?

File: client/extensions/woocommerce/state/site/status/wc-api/test/selectors.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			expect( getLastApiErrorCode( stateWithoutError, 1337 ) ).to.not.exist;
+		} );
+
+		it( 'should not provide an error code if no site data exists.', () => {
+			expect( getLastApiErrorCode( stateWithoutError, 1338 ) ).to.not.exist;
+		} );
+
+		it( 'should match output from getLastApiError.', () => {
+			const err = getLastApiError( stateWithError, 1337 );
+			expect( getLastApiErrorCode( stateWithError, 1337 ) ).to.equal( err.code );
```
Comment (allendav) [inline client/extensions/woocommerce/state/site/status/wc-api/test/selectors.js]: This test doesn't guarantee that you get the code from state, only that both getLastApiError and getLastApiErrorCode return the same value.

File: client/extensions/woocommerce/state/site/status/wc-api/test/selectors.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+			expect( getLastApiErrorCode( stateWithoutError, 1337 ) ).to.not.exist;
+		} );
+
+		it( 'should not provide an error message if no site data exists.', () => {
+			expect( getLastApiErrorCode( stateWithoutError, 1338 ) ).to.not.exist;
+		} );
+
+		it( 'should match output from getLastApiError.', () => {
+			const err = getLastApiError( stateWithError, 1337 );
+			expect( getLastApiErrorMessage( stateWithError, 1337 ) ).to.equal( err.message );
```
Comment (allendav) [inline client/extensions/woocommerce/state/site/status/wc-api/test/selectors.js]: Similar comment as above.

File: client/extensions/woocommerce/state/site/status/wc-api/selectors.js (hunk 1 above)
Comment (coderkevin) [inline client/extensions/woocommerce/state/site/status/wc-api/selectors.js]: I switched to an array for the second parameter.

**Pull request merged**